src/SqlAsyncCollector.cs

```diff
@@ -403,8 +403,11 @@ private static void GenerateDataQueryForMerge(TableInformation table, IEnumerabl
 
             rowData = JsonConvert.SerializeObject(rowsToUpsert, table.JsonSerializerSettings);
             IEnumerable<string> columnNamesFromItem = GetColumnNamesFromItem(rows.First());
-            IEnumerable<string> bracketColumnDefinitionsFromItem = table.Columns.Where(c => columnNamesFromItem.Contains(c.Key))
-                .Select(c => $"{c.Key.AsBracketQuotedString()} {c.Value}");
+            var bracketColumnDefinitionsFromItem = new List<string>();
```

Comment:
Previously `bracketColumnDefinitionsFromItem`'s order was based on the SQL table (`table.Columns`). Instead it should be based on the order of the POCO or JSON item (`columnNamesFromItem`) to match the insert part of the query.